src/ec2/model.ts

```diff
@@ -105,15 +107,17 @@ export class Ec2ConnectionManager {
     }
 
     public async attemptEc2Connection(selection: Ec2Selection): Promise<void> {
+        await this.checkForStartSessionError(selection)
         try {
             const response = await this.ssmClient.startSession(selection.instanceId)
             await this.openSessionInTerminal(response, selection)
         } catch (err) {
-            if (isAwsError(err)) {
-                await this.handleStartSessionError(err, selection)
-            } else {
-                throw err
-            }
+            // Default error if pre-check fails.
+            this.throwConnectionError(
+                'Check that the SSM Agent is running on target instance',
+                'EC2SSMConnect',
+                selection
+            )
```

Comment:
We should add `err` as a `cause` for this new error (it can be added to `ToolkitError`). That way we don't swallow the original error which might contain helpful debugging info.